packages/asset-swapper/src/utils/market_operation_utils/path_optimizer.ts

```diff
@@ -274,7 +274,18 @@ function findRoutesAndCreateOptimalPath(
 
         // TODO(kimpers): remove once we have solved the rounding/precision loss issues in the Rust router
         const maxSampledOutput = BigNumber.max(...routeSamples.map(s => s.output));
-        const scaleOutput = (output: BigNumber) => BigNumber.min(output.times(scale), maxSampledOutput);
+        // Scale output by scale factor but never go above the largest sample (unknown liquidity) or below 1 base unit (unfillable)
+        const scaleOutput = (output: BigNumber) => {
+            // Don't try to scale 0 output as it will be clamped to 1
+            if (output.eq(ZERO_AMOUNT)) {
+                return output;
+            }
+
+            return BigNumber.max(BigNumber.min(output.times(scale), maxSampledOutput), 1).decimalPlaces(
+                0,
+                side === MarketOperation.Sell ? BigNumber.ROUND_FLOOR : BigNumber.ROUND_CEIL,
```

Comment:
This rounding used to be here but was unintentionally removed when `scaleOutput` was rewritten 🙈